### PR TITLE
652: added Document (file) media type

### LINKED ISSUE #652: Add "file" question type - for documents (pdf, doc, odt, ods, xls...)
Hi all,

ODK Collect already supports generic file uploads: https://docs.opendatakit.org/form-question-types/#file-upload-widget

It would be nice to include a new type of question that allows the collector to attach a document (pdf, odt, doc(x), xls(x), odt, ods). Is it of interest or in the roadmap?

### PR DESCRIPTION
**Pull request by viniciuscb**
Proposal for a Document media type.
Captura de tela_2019-11-03_15-04-36

Resolves #652

### CODE REVIEW

**Comment by smoyth** [comment]
Thanks for these PRs! Hoping to get to them next week. In the meantime, if you could resolve any merge conflicts (preferably with a rebase), that would be helpful!

File: app/decorators/odk/question_decorator.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ module Odk
   class QuestionDecorator < ApplicationDecorator
     delegate_all
 
-    URI_DIRS_BY_TYPE = {video: "video", audio: "audio", image: "images"}.freeze
+    URI_DIRS_BY_TYPE = {video: "video", audio: "audio", image: "images", document: "document"}.freeze
```
Comment (smoyth) [inline app/decorators/odk/question_decorator.rb]: If ODK document media prompts aren't supported then this should be removed too.

File: app/decorators/odk/subqing_decorator.rb [hunk 2]
```diff
@@ -127,6 +127,7 @@ def media_type
       when "image", "annotated_image", "sketch", "signature" then "image/*"
       when "audio" then "audio/*"
       when "video" then "video/*"
+      when "document" then "document/*"
```
[the remark this answers is not in the document]
Comment (viniciuscb) [reply app/decorators/odk/subqing_decorator.rb]: Hi @smoyth. Yes, we've tested in ODK collect and also in Nemo web interface. I've tested by uploading a document (pdf, odt, doc) in the collect, the answer was saved and after I could download/preview that document in the answer screen. This also worked in the web interface.

**Comment by smoyth** [comment]
Closing for now due to age. Can be opened again later.

**Pull request closed without merging**